methods_and_recursion/mad_libs.rb

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env ruby
+
+# This helps to check whether 'a' or 'an' will be used.
+class String
+  def vowel?
+    vowels = %w(a e i o u A E I O U)
+    if vowels.include?(self)
+      true
+    else
+      false
+    end
+  end
+end
+
+def ask_for_word(term, list_of_words)
```

Comment:
How about an optional third parameter `seen_before` which defaults to false?

    another_word = seen_before ? 'a' : 'another'